src/script/descriptor.cpp

```diff
@@ -414,31 +419,35 @@ class BIP32PubkeyProvider final : public PubkeyProvider
 
         return true;
     }
-    std::string ToString() const override
+    std::string ToString(bool normalized) const
     {
-        std::string ret = EncodeExtPubKey(m_root_extkey) + FormatHDKeypath(m_path);
+        std::string ret = EncodeExtPubKey(m_root_extkey) + FormatHDKeypath(m_path, /*apostrophe=*/m_apostrophe);
         if (IsRange()) {
             ret += "/*";
-            if (m_derive == DeriveType::HARDENED) ret += '\'';
+            if (m_derive == DeriveType::HARDENED) ret += (!normalized && m_apostrophe) ? '\'' : 'h';
```

Comment:
Looks like it should use `h` everywhere when normalized? (Not only in the hardened-wildcard step.)
```patch
diff --git a/src/script/descriptor.cpp b/src/script/descriptor.cpp
index e3a20c17f4..a2372d7409 100644
--- a/src/script/descriptor.cpp
+++ b/src/script/descriptor.cpp
@@ -421,10 +421,11 @@ public:
     }
     std::string ToString(bool normalized) const
     {
-        std::string ret = EncodeExtPubKey(m_root_extkey) + FormatHDKeypath(m_path, /*apostrophe=*/m_apostrophe);
+        const bool use_apostrophe = !normalized && m_apostrophe;
+        std::string ret = EncodeExtPubKey(m_root_extkey) + FormatHDKeypath(m_path, /*apostrophe=*/use_apostrophe);
         if (IsRange()) {
             ret += "/*";
-            if (m_derive == DeriveType::HARDENED) ret += (!normalized && m_apostrophe) ? '\'' : 'h';
+            if (m_derive == DeriveType::HARDENED) ret += use_apostrophe ? '\'' : 'h';
         }
         return ret;
     }
```